modAionImpl/src/org/aion/zero/impl/db/TransactionStore.java

```diff
@@ -50,23 +56,56 @@ public boolean putToBatch(AionTxInfo tx) {
                 }
             }
             existingInfos.add(tx);
-            source.putToBatch(txHash, existingInfos);
+            txInfoSource.putToBatch(txHash, existingInfos);
 
             return true;
         } finally {
             lock.writeLock().unlock();
         }
     }
 
-    public void flushBatch() {
-        source.flushBatch();
+    public void putAliasesToBatch(AionTxInfo txInfo) {
+        byte[] txHash = txInfo.getReceipt().getTransaction().getTransactionHash();
+
+        lock.writeLock().lock();
+
+        try {
+            for (InternalTransaction itx : txInfo.getInternalTransactions()) {
+                byte[] invokableHash = itx.copyOfInvokableHash();
+
+                if (invokableHash != null) {
+                    Set<ByteArrayWrapper> existingAliases = aliasSource.get(invokableHash);
+
+                    if (existingAliases == null) {
+                        existingAliases = new HashSet<>();
+                    }
+                    existingAliases.add(new ByteArrayWrapper(txHash));
+                    aliasSource.putToBatch(invokableHash, existingAliases);
+                }
+            }
+        } finally {
+            lock.writeLock().unlock();
+        }
+    }
+
+    public void flushTxInfoBatch() {
+        txInfoSource.flushBatch();
+    }
+
+    public void flushAliasBatch() {
+        aliasSource.flushBatch();
+    }
+
+    public void flushAllBatch() {
+        flushTxInfoBatch();
+        flushAliasBatch();
     }
 
-    public AionTxInfo get(byte[] txHash, byte[] blockHash) {
+    public AionTxInfo getTxInfo(byte[] txHash, byte[] blockHash) {
```

Comment:
I would prefer not renaming this to avoid polluting the diff;`getAliases` is sufficient to differentiate the two getters